src/build.rs

```diff
@@ -446,11 +446,26 @@ impl BuildQueue {
             let mut manifest_path = build_dir.clone();
             manifest_path.push("Cargo.toml");
             trace!("manifest_path: {:?}", manifest_path);
+            // TODO: Add support for virtual manifests and multiple packages 
             let ws = Workspace::new(&manifest_path, &config).expect("could not create cargo workspace");
+            let current_package = ws.current().unwrap(); 
+            let targets = current_package.targets(); 
+
+            let bins = {
+                if rls_config.build_bin.is_empty() { 
+                    vec![]
+                } else {
+                    let mut bins = targets.iter().filter(|x| x.is_bin()); 
+                    let bin = bins.find(|x| x.name() == rls_config.build_bin);
+                    match bin { None => vec![], Some(bin) => vec![bin.name().to_owned()]}
```

Comment:
we should warn the user if they specify a bin which we then don't use. Not sure how best to do that - at least we should have a `debug!` message, but maybe we should also fail compilation with a message?